tests/upgrades/test_repository.py

```diff
@@ -89,3 +100,241 @@ def test_post_repository_scenario_upstream_authorization(self):
+    via client then we alter the created custom repository and satellite will be able
+    to sync back the repo.
+
+    Test Steps:
+
+            1. Before Satellite upgrade.
+            2. Create new Organization and Location.
+            3. Create Product, custom repo, cv.
+            4. Create activation key and add subscription in it.
+            5. Create a content host, register and install package on it.
+            6. Upgrade Satellite.
+            7. Remove Old package and add new package into custom repo.
+            8. Sync repo, publish new version of cv.
+            9  Try to install new package on client.
+
+    BZ: 1429201,1698549
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        cls.sat_host = settings.server.hostname
+        cls.docker_vm = os.environ.get('DOCKER_VM')
+        cls.file_path = '/var/www/html/pub/custom_repo/'
+        cls.custom_repo = 'https://{0}{1}'.format(cls.sat_host, '/pub/custom_repo/')
+        _, cls.rpm1_name = os.path.split(rpm1)
+        _, cls.rpm2_name = os.path.split(rpm2)
+
+    def _check_package_installed(self, client_container_id, package):
+        """Verify if package is installed on docker content host.
+
+        :param: str client_container_id: Container ID
+        :param: str package : Package name
+        """
+        kwargs = {'host': self.docker_vm}
+        installed_package = execute(
+            docker_execute_command,
+            client_container_id,
+            'rpm -qa|grep {}'.format(package),
+            **kwargs
+        )[self.docker_vm]
+        self.assertIn(package, installed_package)
+
+    def _create_repo(self, post_upgrade=False):
+        """ Creates a custom yum repository, that will be synced to satellite
+        """
+        if post_upgrade:
+            run('wget {0} -P {1}'.format(rpm2, self.file_path))
+            run('rm -rf {0}'.format(self.file_path + self.rpm1_name))
+            run('createrepo --update {0}'.format(self.file_path))
+        else:
+            run('mkdir /var/www/html/pub/custom_repo')
+            run('wget {0} -P {1}'.format(rpm1, self.file_path))
+            run('createrepo --database {0}'.format(self.file_path))
+
+    def _install_package(self, client_container_id, package):
+        """Install the package on docker content host.
+
+        :param: str client_container_id: Container ID
+        :param: str package : package name
+        """
+        rpm_name = package.split('-')[0]
+        kwargs = {'host': self.docker_vm}
+        execute(docker_execute_command,
+                client_container_id,
+                'subscription-manager repos;yum clean all',
+                **kwargs)[self.docker_vm]
+        command = 'yum install -y {}'.format(rpm_name)
+        execute(docker_execute_command, client_container_id, command, **kwargs)[self.docker_vm]
+        self._check_package_installed(client_container_id, rpm_name)
+
+    def _host_status(self, client_container_name=None):
```

Comment:
Why do we need function for this ? You can directly use it.

Reply:
I am using same at mutiple time under  `_host_location_update` hence used in this way.